pandas/tests/indexes/test_base.py

```diff
@@ -302,138 +290,138 @@ def test_index_ctor_infer_periodindex(self):
+        None, 'UTC', 'US/Eastern', 'Asia/Tokyo'])
+    @pytest.mark.parametrize("values", [
+        # pass values without timezone, as DatetimeIndex localizes it
+        pd.date_range('2011-01-01', periods=5).values,
+        pd.date_range('2011-01-01', periods=5).asi8])
+    @pytest.mark.parametrize("klass", [pd.Index, pd.DatetimeIndex])
+    def test_constructor_dtypes_datetime(self, tz, values, klass):
+        idx = pd.date_range('2011-01-01', periods=5, tz=tz)
+        dtype = idx.dtype
 
-            # pass values without timezone, as DatetimeIndex localizes it
-            for values in [pd.date_range('2011-01-01', periods=5).values,
-                           pd.date_range('2011-01-01', periods=5).asi8]:
+        res = klass(values, tz=tz)
+        tm.assert_index_equal(res, idx)
 
-                for res in [pd.Index(values, tz=tz),
-                            pd.Index(values, dtype=dtype),
-                            pd.Index(list(values), tz=tz),
-                            pd.Index(list(values), dtype=dtype)]:
-                    tm.assert_index_equal(res, idx)
+        res = klass(values, dtype=dtype)
+        tm.assert_index_equal(res, idx)
 
-                # check compat with DatetimeIndex
-                for res in [pd.DatetimeIndex(values, tz=tz),
-                            pd.DatetimeIndex(values, dtype=dtype),
-                            pd.DatetimeIndex(list(values), tz=tz),
-                            pd.DatetimeIndex(list(values), dtype=dtype)]:
-                    tm.assert_index_equal(res, idx)
+        res = klass(list(values), tz=tz)
+        tm.assert_index_equal(res, idx)
 
-    def test_constructor_dtypes_timedelta(self):
+        res = klass(list(values), dtype=dtype)
+        tm.assert_index_equal(res, idx)
 
+    @pytest.mark.parametrize("attr", ['values', 'asi8'])
+    @pytest.mark.parametrize("klass", [pd.Index, pd.TimedeltaIndex])
+    def test_constructor_dtypes_timedelta(self, attr, klass):
         idx = pd.timedelta_range('1 days', periods=5)
         dtype = idx.dtype
 
-        for values in [idx.values, idx.asi8]:
+        values = getattr(idx, attr)
 
-            for res in [pd.Index(values, dtype=dtype),
-                        pd.Index(list(values), dtype=dtype)]:
-                tm.assert_index_equal(res, idx)
+        res = klass(values, dtype=dtype)
+        tm.assert_index_equal(res, idx)
 
-            # check compat with TimedeltaIndex
-            for res in [pd.TimedeltaIndex(values, dtype=dtype),
-                        pd.TimedeltaIndex(list(values), dtype=dtype)]:
-                tm.assert_index_equal(res, idx)
+        res = klass(list(values), dtype=dtype)
+        tm.assert_index_equal(res, idx)
 
-    def test_constructor_empty(self):
+    def test_constructor_empty_gen(self):
         skip_index_keys = ["repeats", "periodIndex", "rangeIndex",
                            "tuples"]
         for key, idx in self.generate_index_types(skip_index_keys):
```

Comment:
In general some of these magic class methods could probably be replaced with pytest functionality. I haven't looked in too much detail just yet but was planning on reviewing after giving the module a first pass at parametrization without changing class behavior

Reply:
yes see my comments below

Reply:
note there is already a fixture of ``indices`` which works, so have 3 methods of specifying things ATM:

- ``indices`` fixture
- ``generate_index_types(...)``
- ``self.indices``

needs to clean this and just make fixtures that we can use generally (in conftest) with docs